File: chap3/julia/code4.jl
using Flux

## スキップコネクションを持つモデルの定義
z1 = Dense(784, 256, relu)
z2 = Dense(256, 128, relu)
z3 = Dense(128, 256, relu)
skip = SkipConnection(z1, +)
y = Dense(128, 10)
model = Chain(
    z1, 
    z2, 
    z3, 
    skip, 
    y, 
    softmax
)

File: chap3/R/code4.r
## スキップコネクションを持つモデルの定義
x = layer_input(784)
z1 = x %>% layer_dense(
    units = 256,
    activation = "relu"
)
z2 = z1 %>% layer_dense(
    units = 128,
    activation = "relu"
)
z3 = z2 %>% layer_dense(
    units = 256,
    activation = "relu"
)
skip = layer_add(c(z1, z3))
y = z3 %>% layer_dense(
    units = 10,
    activation = "softmax"
)
model = keras_model(x, y)

model # modelの出力
